src/main/java/FlyChat/Core/TaskList.java

```diff
@@ -0,0 +1,186 @@
+        storageList.get(index - 1).completeTask();
+        return "I've marked the task as done ^o^ :\n  " + storageList.get(index - 1).toString();
+    }
+
+    /**
+     * Marks the task specified by the task index as not completed.
+     *
+     * @return String announces command completion.
+     * @throws IndexOutOfBoundsException If task index is out of bounds of the task list.
+     */
+    public String unmark(int index) throws IndexOutOfBoundsException {
+        //index - 1 because task list displayed to user starts from 1
+        storageList.get(index - 1).uncompleteTask();
+        return "I've marked the task as not done T.T :\n  " + storageList.get(index - 1).toString();
+    }
+
+    /**
+     * Deletes the task specified by the task index from the task list.
+     *
+     * @return String announcing task completion.
+     * @throws IndexOutOfBoundsException If task index is out of bounds of the task list.
+     */
+    public String deleteTask(int index) throws IndexOutOfBoundsException {
+        Task targetTask = storageList.get(index - 1);
+        storageList.remove(targetTask);
+        return "Task removed:\n  " + targetTask + "\nNow you have " + storageList.size()
+                + " tasks in the list. SOLDIER ON! ^o^";
+    }
+
+    /**
+     * Finds tasks that contain the corresponding key word in their descriptions.
+     *
+     * @param keyWord
+     * @return String announcing results of the search.
+     */
+    public String find(String keyWord) {
+        int counter = 1;
+        String finalString = "Here are the matching tasks in your list:\n";
+
+        if (keyWord.equals("")) {
+            return finalString;
+        }
+
+        for (Task t : storageList) {
+            if (t.getDescription().contains(keyWord)) {
+                finalString += counter + ". " + t.toString() + "\n";
+                counter++;
+            }
+        }
+
+        return finalString;
+    }
+
+    /**
+     * Reads the task from the save file and adds it to the task list.
+     *
+     * @param str Line from save file containing task information.
+     * @throws IOException If save file is corrupted and unable to be read.
+     */
+    public void addTaskFromFile(String str) throws IOException {
+        boolean isCompleted = false;
+        isCompleted = parser.checkTaskCompletedFromFile(str);
+
+        //format the task string properly using regex
+        if (parser.getTaskTypeFromFile(str).equals("[T]")) {
+            //To-do case
+            addToList(Todo.createNewTodo(parser.getTaskDescriptionFromFile(str), isCompleted));
+        } else if (parser.getTaskTypeFromFile(str).equals("[D]")) {
+            //Deadline case
+            addToList(Deadline.createNewDeadline(parser.getTaskDescriptionFromFile(str),
+                    parser.getDeadlineEndDateFromFile(str), isCompleted));
+        } else if (parser.getTaskTypeFromFile(str).equals("[E]")) {
```

Comment:
Maybe `getTaskTypeFromFile` should just return `T`, `D`, or `E` without the brackets. 
Just a minor nitpick.